modules/ximgproc/include/opencv2/ximgproc/slic.hpp

```diff
@@ -138,21 +140,22 @@ class CV_EXPORTS_W SuperpixelSLIC : public Algorithm
 
 @param image Image to segment
 @param algorithm Chooses the algorithm variant to use:
-SLIC segments image using a desired region_size, and in addition
-SLICO will choose an adaptive compactness factor.
+SLIC segments image using a desired region_size, and in addition SLICO will optimize using adaptive compactness factor,
+while MSLIC will optimize using manifold methods resulting in more content-sensitive superpixels.
 @param region_size Chooses an average superpixel size measured in pixels
 @param ruler Chooses the enforcement of superpixel smoothness factor of superpixel
 
 The function initializes a SuperpixelSLIC object for the input image. It sets the parameters of choosed
 superpixel algorithm, which are: region_size and ruler. It preallocate some buffers for future
-computing iterations over the given image. An example of SLIC versus SLICO is ilustrated in the
-following picture.
+computing iterations over the given image. For enanched results it is recommended for color images to
+preprocess image with little gaussian blur using a small 3 x 3 kernel and additional conversion into
+CieLAB color space. An example of SLIC versus SLICO and MSLIC is ilustrated in the following picture.
 
 ![image](pics/superpixels_slic.png)
 
  */
 
-    enum SLIC { SLIC = 100, SLICO = 101 };
+    enum SLIC { SLIC = 100, SLICO = 101, MSLIC = 102 };
```

Comment:
Please, move this enum to the line 63 in this file to let Doxygen process it correctly.
Now description of the `createSuperpixelSLIC` function is empty.